src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/services/GeneratePdfAccreditedProgrammesServiceTest.kt

```diff
@@ -0,0 +1,99 @@
+package uk.gov.justice.digital.hmpps.subjectaccessrequestworker.services
+
+import com.itextpdf.kernel.pdf.PdfDocument
+import com.itextpdf.kernel.pdf.PdfReader
+import com.itextpdf.kernel.pdf.PdfWriter
+import com.itextpdf.kernel.pdf.canvas.parser.PdfTextExtractor
+import com.itextpdf.layout.Document
+import com.microsoft.applicationinsights.TelemetryClient
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.models.DpsService
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.utils.TemplateHelpers
+import java.io.FileOutputStream
+
+class GeneratePdfAccreditedProgrammesServiceTest {
+  private val templateHelpers = TemplateHelpers()
+  private val templateRenderService = TemplateRenderService(templateHelpers)
+  private val telemetryClient: TelemetryClient = mock()
+  private val generatePdfService = GeneratePdfService(templateRenderService, telemetryClient)
+
+  @Test
+  fun `generatePdfService renders for Accredited Programmes Service`() {
+    val serviceList =
+      listOf(DpsService(name = "hmpps-accredited-programmes-api", content = accreditedProgrammesServiceData))
+    val pdfDocument = PdfDocument(PdfWriter(FileOutputStream("dummy-accredited-programmes-template.pdf")))
```

Comment:
Might be worth using `@TempDir` so its automatically deleted after each test run.

https://www.baeldung.com/junit-5-temporary-directory

Reply:
on `@Tempdir` - we need that file